src/app/overview/deposit-view/deposit-view.component.ts

```diff
@@ -0,0 +1,174 @@
+import { Component, OnInit } from '@angular/core';
+import { ExplorerModel } from '../../explorer/explorer.model';
+import { ExplorerService } from '../../explorer/explorer.service';
+import { WalletModel } from '../../wallet/wallet.model';
+import { WalletService } from '../../wallet/wallet.service';
+import { RpcSend } from '../../rpc/rpc-send.model';
+import { RpcReceive } from '../../rpc/rpc-receive.model';
+
+export interface SendToAddressModel {
+  amount: Number;
+  destinationAddress: String;
+  feeIncluded: Boolean;
+}
+
+
+@Component({
+  selector: 'app-deposit-view',
+  templateUrl: './deposit-view.component.html',
+  styleUrls: ['../overview.component.css','./deposit-view.component.css']
+})
+export class DepositViewComponent implements OnInit {
+
+  explorer: ExplorerModel;
+  wallet: WalletModel;
+  transaction: SendToAddressModel = {
+    amount: undefined,
+    destinationAddress: undefined,
+    feeIncluded: false,
+  };
+  rpcReceive: RpcReceive;
+  qrMainAddress: string;
+
+  constructor(
+    private explorerService: ExplorerService,
+    private walletService: WalletService
+  ) {}
+
+  ngOnInit() {
```

Comment:
I think all we need on this component at the moment is the functionality that adds `mainAddress` to `this.wallet` all the other functions can go.